language/move-compiler/src/parser/syntax.rs

```diff
@@ -1718,6 +1727,15 @@ fn parse_type(context: &mut Context) -> Result<Type, Box<Diagnostic>> {
     Ok(spanned(context.tokens.file_hash(), start_loc, end_loc, t))
 }
 
+/// Checks whether the next tokens looks like the start of a type. Must be aligned
+/// with `parse_type`.
+fn is_start_of_type(context: &mut Context) -> bool {
+    matches!(
+        context.tokens.peek(),
+        Tok::LParen | Tok::Amp | Tok::AmpMut | Tok::Pipe | Tok::Identifier
```

Comment:
This might easily go out of sync with `parse_type`, for example, `Tok::NumValue` is also possible as a start of type `0x1::MyModule::MyStruct`.

Reply:
... which is what the comment also says. Feels we do not change the syntax that often that this would be an issue.